Modules/IO/DCMTK/src/itkDCMTKFileReader.cxx

```diff
@@ -1250,26 +1253,38 @@ DCMTKFileReader::GetSpacing(double * const spacing) const
            *  when we can not get (0018, 0088), we should compute spacing
            * from the planes' positions (TODO, see PR 112).
            * */
-          if (subSequence.GetElementDS<double>(0x0028, 0x0030, 2, _spacing, false) == EXIT_SUCCESS)
+          if (!(_have_val[0] && _have_val[1]))
           {
-            spacing[0] = _spacing[1];
-            spacing[1] = _spacing[0];
-            if (subSequence.GetElementDS<double>(0x0018, 0x0088, 1, &_spacing[2], false) == EXIT_SUCCESS)
+            if (subSequence.GetElementDS<double>(0x0028, 0x0030, 2, _spacing, false) == EXIT_SUCCESS)
             {
-              spacing[2] = _spacing[2];
+              spacing[0] = _spacing[1];
+              spacing[1] = _spacing[0];
+              _have_val[0] = true;
+              _have_val[1] = true;
             }
-            else
+          }
+          if (!_have_val[2])
+          {
+            if (subSequence.GetElementDS<double>(0x0018, 0x0088, 1, &_spacing[2], false) == EXIT_SUCCESS)
             {
-              // punt, zSpace of 1
-              spacing[2] = 1.0;
+              spacing[2] = _spacing[2];
+              _have_val[2] = true;
             }
-            break;
+          }
+          if (_have_val[0] && _have_val[1] && _have_val[2])
+          {
+            return EXIT_SUCCESS;
           }
         }
       }
     }
   }
-  return rval;
+
+  if (_have_val[0] && _have_val[1] && _have_val[2])
+  {
+    return EXIT_SUCCESS;
+  }
+  return EXIT_FAILURE;
```

Comment:
Maybe initialize `_spacing[3]` with `1.0, 1.0, 1.0` and before `return EXIT_FAILURE;` add
```
 spacing[0] = _spacing[0];
 spacing[1] = _spacing[1];
 spacing[2] = _spacing[2];
```
The return value of the function does not seem to be checked, and the double[3] pointer passed to the function is not initialized, so it should be safer. Spacing `0` usually triggers exception later, IMHO, the default is usually `1`.
https://github.com/InsightSoftwareConsortium/ITK/blob/16f42267a00e2c2ae35c189609fad3147ce99e67/Modules/IO/DCMTK/src/itkDCMTKImageIO.cxx#L436

In fact, BTW, I am afraid that support for enhanced multi-frames IODs is rather basic.

Reply:
> BTW, I am afraid that support for enhanced multi-frame IODs is rather basic.

This is going to be increasingly important.  I've heard that Siemens MR systems will start creating multiframe instances by default soon.